glue/sample/src/sinter/_sampling_sampler_class.py

```diff
@@ -0,0 +1,123 @@
+from typing import Tuple
+import abc
+import pathlib
+
+import numpy as np
+import stim
+
+
+class CompiledSampler(metaclass=abc.ABCMeta):
+    """Abstract class for samplers preconfigured to a specific sampling task.
+
+    This is the type returned by `sinter.Sampler.compile_sampler_for_circuit`. The
+    idea is that, when many shots of the same sampling task are going to be
+    performed, it is valuable to pay the cost of configuring the sampler only
+    once instead of once per batch of shots. Custom samplers can optionally
+    implement that method, and return this type, to increase sampling
+    efficiency.
+    """
+    @abc.abstractmethod
+    def sample_detectors_bit_packed(
```

Comment:
I think, instead of sampling detection events, it would be better to sample `AnonTaskStats` directly. This would allow for much more flexibility in what a sampler did. But then I'm not sure how the decoder exactly fits into the picture... is it an argument to this method? Bleh...

Reply:
Thanks for the review! Could you give a usecase where sampling `AnnoTaskStats` will help? I'm not sure how to settle down the interface yet since it will introduce strong coupling between `Sampler` and `Decoder` if I feed the decoder as an argument to the sampling method.